File: golangServer/modules/ModuleRegistry.go
package modules

import (
	// "os/exec"
	// "os"
	// "log"
	// "github.com/BurntSushi/toml"
	// "google.golang.org/grpc"
)

type ModuleRegistry struct {
	modules []IModule
	activeModules []IModule
}

/////////////////////////
// Module Registration //
/////////////////////////
func (registry *ModuleRegistry) RegisterModules() {

	var minimegaConfig MinimegaConfig
	var minimegaProtobufServer MinimegaProtobufServer

	minimega := &MinimegaModule{ 
		ModuleContext {
			Reference: "minimega",
		},
		minimegaConfig,
		minimegaProtobufServer,
		false,
	}

	registry.modules = append(registry.modules, minimega)
}

func (registry *ModuleRegistry) GetRegisteredModules() []IModule {
	return registry.modules
}

func (registry *ModuleRegistry) GetActiveModules() []IModule {

	for _, module := range registry.modules {
		if module.isActive() {
			registry.activeModules = append(registry.activeModules, module)
		}
	}
	
	return registry.activeModules
}

File: Client/Modules/ModuleRegistry.cs
using System;
using System.Collections.Generic;
using Client.Modules.Minimega;

namespace Client.Modules
{
    public class ModuleRegistry
    {
        private List<Module> _registeredModules { get; set; }

        public void RegisterModules() {
            
            _registeredModules = new List<Module>();

            _registeredModules.Add(
                new MinimegaModule()
            );
        }

        internal List<Module> GetRegisteredModules()
        {
            return _registeredModules;
        }
    }
}